core/data/src/main/java/com/puzzle/data/billing/BillingHelperImpl.kt

```
@@ -0,0 +1,163 @@
+        if (billingClient.isReady) {
+            continuation.resume(Result.success(Unit))
+            return@suspendCancellableCoroutine
+        }
+
+        billingClient.startConnection(object : BillingClientStateListener {
+            override fun onBillingSetupFinished(result: BillingResult) {
+                if (result.responseCode == BillingResponseCode.OK) continuation.resume(Unit)
+                else continuation.resumeWithException(Exception("Billing setup failed: ${result.debugMessage}"))
+            }
+
+            override fun onBillingServiceDisconnected() {}
```

detekt (1.23.8)

[warning] This empty block of code can be removed.

(detekt.empty-blocks.EmptyFunctionBlock)